### DISCUSSION TITLE
Automatically infering ResourceCollection resource

### DISCUSSION THREAD

**Discussion by ezraanglo** [dedoc/scramble#508] [Q&A]
Hi! Thank you for this amazing tool!

I'm restarting my laravel project and using the latest releases, decided to implement swagger and saw a post about scramble so I gave it a try! I'm just having a problem with automatically infering the "data" of my resource when im doing a collection.

Basically when I do new AnonymousResourceCollection($data, UserResource::class), its not infering my resource type. 

But if I create a new UserCollection class, it works. Is there a way to work around this without creating a new file and class?

Thank you!

**Comment by romalytvynenko** [comment]
Hey @ezraanglo please share real samples from your codebase

**Reply by ezraanglo** [reply]
Thank you for prompt response!

UserResource.php
```
<?php

namespace App\Http\Resources;

use Illuminate\Http\Request;

class UserResource extends ApiResource
{
    public function toArray(Request $request): array
    {
        return [
            'id' => $this->id,
            'name' => $this->name,
            'email' => $this->email,=
        ];
    }
}

```

ResourceCollection.php
```
<?php

namespace App\Http\Resources;

use Illuminate\Http\Request;
use Illuminate\Http\Resources\Json\ResourceCollection as JsonResourceCollection;

class ResourceCollection extends JsonResourceCollection
{
    public function __construct($resource, $collects = null)
    {
        parent::__construct($resource);
        $this->collects = $collects;
    }

    /**
     * Transform the resource collection into an array.
     *
     * @return array<int|string, mixed>
     */
    public function toArray(Request $request): array
    {
        return [
            'success' => true,
            'status' => 200,
            'message' => 'Success',
            'total' => $this->count(),
            'data' => $this->collection
        ];
    }
}

```

UserCollection.php
```
<?php

namespace App\Http\Resources;

class UserCollection extends ResourceCollection
{

}

```

Then on my controller, if I do
```
return new UserCollection($data)
```
it returns
```
{
  "success": true,
  "status": 200,
  "message": "Success",
  "total": "string",
  "data": [
    {
      "id": 0,
      "name": "string",
      "email": "string",
    }
  ]
}
```

But if i do 
```
return new ResourceCollection($data, UserResource::class);
```
it returns
```
{
  "success": true,
  "status": 200,
  "message": "Success",
  "total": "string",
  "data": "string"
}
```

**Reply by romalytvynenko** [reply]
Oh wow, that's really custom. But I think it won't be that hard to make this work. I'll look into it, not sure about eta

**Reply by ezraanglo** [reply]
Thank you! Its my first time using Resources with laravel so i don't know if im customizing it right, but I really just want to wrap my responses (item and collection) with 

```
return [
    'success' => true,
    'status' => 200,
    'message' => 'Success',
    'total' => $this->count(),
    'data' => $this->collection
];
```

If you think there is a better way of doing this that scramble already supports, im also open to refactoring. Again thank you!
